flink-tests/src/test/java/org/apache/flink/runtime/operators/lifecycle/graph/TestEventSource.java

```diff
@@ -103,6 +110,14 @@ private void ack(TestCommand cmd) {
 
     @Override
     public void cancel() {
+        stop();
+    }
+
+    private void stop() {
+        commandQueue.unsubscribe(operatorID, commandExecutor);
         isRunning = false;
+        if (!scheduledCommands.isEmpty()) {
+            LOG.warn("Unsubscribed with remaining commands: {}", scheduledCommands);
```

Comment:
If this happens, is still possible to time out (in `TestJobExecutor#waitForFailover`) while waiting for failover, because we haven't acknowledged the FAIL command?

Reply:
Yes, it is possible currently.
I'm adding dec05f98e84be1007aa2c6471b9e767a7f1ca235 to make this impossible (process all scheduled commands).